chalice/awsclient.py

```diff
@@ -75,10 +78,17 @@ def create_function(self, function_name, role_arn, zip_contents):
                 continue
             return response['FunctionArn']
 
-    def update_function_code(self, function_name, zip_contents):
-        # type: (str, str) -> Dict[str, Any]
-        return self._client('lambda').update_function_code(
+    def update_function(self, function_name, zip_contents,
+                        environment_variables=None):
+        # type: (str, str, Optional[Dict[str, str]]) -> Dict[str, Any]
+        lambda_client = self._client('lambda')
+        return_value = lambda_client.update_function_code(
             FunctionName=function_name, ZipFile=zip_contents)
+        if environment_variables is not None:
```

Comment:
The way this code sits currently won't delete environment variables if the user removes all references to it from their config file. I think it makes sense to have the deployment reflect the config as much as possible. Maybe use something like the following if this is functionality that you want.
`
if not environment_variables:
    environment_variables = {}
lambda_client.update_function_configuration(
    FunctionName=function_name,
    Environment={"Variables": environment_variables})
`